### ISSUE TITLE
Frequent (95% of the time) "Cannot find context with specified id" pyppeteer error during "Run backup" GitHub action step

### ISSUE THREAD

**Issue by tobek** [MatthieuBizien/roam-to-git#87]
**Describe the bug**

I set roam-to-git up a few days ago using GitHub actions. It worked the first time, and since then about 19/20 times that the GitHub action runs, it fails on the "Run backup" step. I am not using a Google account. I don't have any pages with long titles or titles starting with `/`.

A quick google for the `pyppeteer` error message seems like it has to do with a page loading slowly and the code not retrying?

**To Reproduce**

- Set up GitHub action, let it run

**Traceback**

```
Run roam-to-git --skip-git .
  roam-to-git --skip-git .
  shell: /bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.8.6/x64
    ROAMRESEARCH_USER: ***
    ROAMRESEARCH_PASSWORD: ***
    ROAMRESEARCH_DATABASE: ***
2021-01-04 20:14:28.399 | DEBUG    | roam_to_git.__main__:main:53 - No secret found at /home/runner/work/roam-***/roam-***/.env
2021-01-04 20:14:28.405 | DEBUG    | roam_to_git.scrapping:download_rr_archive:55 - Creating browser
[W:pyppeteer.chromium_downloader] start chromium download.
Download may take a few minutes.

  0%|          | 0/108773488 [00:00<?, ?it/s]
  2%|▏         | 2621440/108773488 [00:00<00:04, 26209550.90it/s]
 15%|█▌        | 16783360/108773488 [00:00<00:01, 68414351.01it/s]
 31%|███       | 33525760/108773488 [00:00<00:00, 105953651.53it/s]
 48%|████▊     | 52602880/108773488 [00:00<00:00, 135718383.51it/s]
 65%|██████▍   | 70297600/108773488 [00:00<00:00, 149478185.65it/s]
 79%|███████▉  | 85749760/108773488 [00:00<00:00, 148025058.14it/s]
 96%|█████████▌| 104663040/108773488 [00:00<00:00, 160540803.51it/s]
100%|██████████| 108773488/108773488 [00:00<00:00, 137014080.98it/s]
[W:pyppeteer.chromium_downloader] 
chromium download done.
[W:pyppeteer.chromium_downloader] chromium extracted to: /home/runner/.local/share/pyppeteer/local-chromium/588429
2021-01-04 20:14:32.266 | DEBUG    | roam_to_git.scrapping:download_rr_archive:55 - Creating browser
2021-01-04 20:14:32.823 | DEBUG    | roam_to_git.scrapping:_download_rr_archive:92 - Configure downloads to /tmp/tmp50gtgktn
2021-01-04 20:14:32.835 | DEBUG    | roam_to_git.scrapping:_download_rr_archive:92 - Configure downloads to /tmp/tmpm_g64a51
2021-01-04 20:14:32.869 | DEBUG    | roam_to_git.scrapping:signin:185 - Opening signin page
2021-01-04 20:14:32.881 | DEBUG    | roam_to_git.scrapping:signin:185 - Opening signin page
2021-01-04 20:14:37.346 | DEBUG    | roam_to_git.scrapping:signin:189 - Fill email '***'
2021-01-04 20:14:37.675 | DEBUG    | roam_to_git.scrapping:signin:189 - Fill email '***'
2021-01-04 20:14:38.307 | DEBUG    | roam_to_git.scrapping:download_rr_archive:76 - Closing browser markdown
2021-01-04 20:14:40.629 | DEBUG    | roam_to_git.scrapping:download_rr_archive:78 - Closed browser markdown
2021-01-04 20:14:40.629 | ERROR    | __main__:<module>:33 - An error has been caught in function '<module>', process 'MainProcess' (2826), thread 'MainThread' (140694709565248):
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/pyppeteer/execution_context.py", line 80, in evaluateHandle
    _obj = await self._client.send('Runtime.evaluate', {
                 │    │       └ <function CDPSession.send at 0x7ff60324baf0>
                 │    └ <pyppeteer.connection.CDPSession object at 0x7ff602730790>
                 └ <pyppeteer.execution_context.ExecutionContext object at 0x7ff6020a14f0>

pyppeteer.errors.NetworkError: Protocol error (Runtime.evaluate): Cannot find context with specified id


During handling of the above exception, another exception occurred:

[...]
```

**Run `roam-to-git --debug notes/` and report what you get.**

I haven't set this up locally yet, I will try.

**Please complete the following information:**

- OS: Linux but it doesn't matter because
- Do you use Github Action? Yes
- Do you use multiple Roam databases? Just 1 graph
- Does roam-to-git use to work for you? When precisely did it stopped to work? Started using it 3 days ago, it has run every hour, and only worked 4 times.
- Does some backup runs are still working? Yeah

**Comment by bradlindblad** [comment]
same issue here.

**Comment by adithyabsk** [comment]
I also have been getting many failed runs which all fail on `email_elem = await document.querySelector("input[name='email']")` in `scrapping.py`.

https://github.com/MatthieuBizien/roam-to-git/blob/7af45367622c4547cf53cfb8b47b2d1931de580e/roam_to_git/scrapping.py#L183-L203

I spent some time debugging this issue and I think I am able to reproduce the problem:

https://user-images.githubusercontent.com/4440808/105118192-eb5b6c00-5a9b-11eb-94d7-269e2bb4e527.mp4

When I was running the program in debug mode I noticed that Roam seems to sometimes refresh a second time on the login screen. I could not reproduce this in a normal browser instance (i.e. non-puppeteer). Instead of having to increase the sleep-duration everywhere, I have manually increased the sleep duration for just this particular step to around 5 seconds which seems to have done the trick for me.

Not sure of a bulletproof solution here and am curious if other people are observing this extra refresh. I'm going to observe how this patch performs on my fork over the next few days.
